salt/_states/containerd.py

```diff
@@ -0,0 +1,118 @@
+'''
+States to manage the :program:`containerd` CRI runtime.
+'''
+
+import logging
+
+log = logging.getLogger(__name__)
+
+
+__virtualname__ = 'containerd'
+
+
+def __virtual__():
+    if 'cri.list_images' not in __salt__:
+        return (False, 'Missing \'cri\' module')
+    if 'containerd.load_cri_image' not in __salt__:
+        return (False, 'Missing \'containerd\' module')
+
+    return __virtualname__
+
+
+def image_managed(name, archive_path=None):
+    '''
+    Pull or load an image in the CRI image cache.
+
+    name
+        Tag or digest of the image
+    archive_path : None
+        Optional local Docker archive of the image to load if image not present.
+        If this is set, no pull will be performed.
+    '''
+
+    ret = {
+        'name': name,
+        'result': False,
+        'changes': {},
+        'pchanges': {},
+        'comment': '',
+    }
+
+    images = __salt__['cri.list_images']()
+    if images is None:
+        ret['comment'] = 'Failed to list CRI images'
+        return ret
+
+    available = False
```

Comment:
Minor: could be rewritten

```python
    for image in images:
        if name in image.get('repoTags', []):
            available = True
            break
        if name in image.get('repoDigests', []):
            available = True
            break
    else:
        available = False
```

Reply:
I don't like that approach, more of a "Define with failure value, then maybe turn into success". Also, I vehemently dislike the `for/else` construction in Python, which is non-obvious for anyone except those who are familiar with this obscure feature.

Reply:
Fair enough, yeah it's really Python-specific